src/bun.js/node/win_watcher.zig

```diff
@@ -0,0 +1,391 @@
+const StoredFileDescriptorType = bun.StoredFileDescriptorType;
+const Output = bun.Output;
+
+var default_manager: ?*WinPathWatcherManager = null;
+
+// TODO: make this a generic so we can reuse code with path_watcher
+// TODO: we probably should use native instead of libuv abstraction here for better performance
+pub const WinPathWatcherManager = struct {
+    const GenericWatcher = @import("../../watcher.zig");
+    const options = @import("../../options.zig");
+    pub const Watcher = GenericWatcher.NewWatcher(*WinPathWatcherManager);
+    const log = Output.scoped(.WinPathWatcherManager, false);
+    main_watcher: *Watcher,
+
+    watchers: bun.BabyList(?*PathWatcher) = .{},
+    watcher_count: u32 = 0,
+    vm: *JSC.VirtualMachine,
+    file_paths: bun.StringHashMap(PathInfo),
+    deinit_on_last_watcher: bool = false,
+    const PathInfo = struct {
+        fd: StoredFileDescriptorType = .zero,
+        is_file: bool = true,
+        path: [:0]const u8,
+        dirname: string,
+        refs: u32 = 0,
+        hash: Watcher.HashType,
+    };
+
+    fn _fdFromAbsolutePathZ(
+        this: *WinPathWatcherManager,
+        path: [:0]const u8,
+    ) !PathInfo {
+        if (this.file_paths.getEntry(path)) |entry| {
+            var info = entry.value_ptr;
+            info.refs += 1;
+            return info.*;
+        }
+        const cloned_path = try bun.default_allocator.dupeZ(u8, path);
+        errdefer bun.default_allocator.free(cloned_path);
+
+        if (std.fs.openDirAbsoluteZ(cloned_path, .{
+            .access_sub_paths = true,
+            .iterate = true,
+        })) |iterable_dir| {
+            const result = PathInfo{
+                .fd = bun.toFD(iterable_dir.fd),
+                .is_file = false,
+                .path = cloned_path,
+                .dirname = cloned_path,
+                .hash = Watcher.getHash(cloned_path),
+                .refs = 1,
+            };
+            _ = try this.file_paths.put(cloned_path, result);
+            return result;
+        } else |err| {
+            if (err == error.NotDir) {
+                const file = try std.fs.openFileAbsoluteZ(cloned_path, .{ .mode = .read_only });
+                const result = PathInfo{
+                    .fd = bun.toFD(file.handle),
+                    .is_file = true,
+                    .path = cloned_path,
+                    // if is really a file we need to get the dirname
+                    .dirname = std.fs.path.dirname(cloned_path) orelse cloned_path,
+                    .hash = Watcher.getHash(cloned_path),
+                    .refs = 1,
+                };
+                _ = try this.file_paths.put(cloned_path, result);
+                return result;
+            } else {
+                return err;
+            }
+        }
+
+        unreachable;
+    }
+
+    pub fn init(vm: *JSC.VirtualMachine) !*WinPathWatcherManager {
+        const this = try bun.default_allocator.create(WinPathWatcherManager);
```

Comment:
This should use `bun.New` instead

Reply:
fixed in https://github.com/oven-sh/bun/pull/8379/commits/b323bc66cfc1bb68130489c1f03e90679f6667a2

Reply:
Oh wait a min I used `bun.new` instead of `bun.New` let me fix it

Reply:
Ok now is using `bun.New` https://github.com/oven-sh/bun/commit/b602df9190241aab2bb0074985b4b7d1599340df